applications/plexos-hpc-walkthrough/Run-PLEXOS.md

```diff
@@ -1,69 +1,205 @@
+[here](https://github.nrel.gov/hsorense/plexos-smoke-test) on the internal
+GitHub website. We should be able to use the following commands to run the test
+
+```bash
+cd /scratch/$USER
+mkdir smoke
+cd smoke
+git clone https://github.nrel.gov/hsorense/plexos-smoke-test
+cd plexos-smoke-test/
+
+sbatch --account=MYACCOUNT submit_8.0.sh
 ```
 
-6. Run PLEXOS
+Remeber to replace `MYACCOUNT` with your own HPC account name. once the job has
+been submitted, it is possible to ssh into the compute node and view what is happening
+on the node. Something like the following can be used
+
+```bash
+squeue -u $USER
+# Get the node on which your job running
+ssh $NODENAME
+ps -U $USER -L -o pid,lwp,psr,comm,pcpu | grep -v COMMAND | sort -k3n
+```
+
+## Example 2: MSPCM Workshop-OneWeek
+
+This example runs a base case that simulates one week of the energy market
 
+1. Clone the MSPCM Workshop and move into the model directory
   ```bash
-mono $PLEXOS/PLEXOS64.exe -n "one_week_model.xml" -m DAY_AHEAD
+  cd /scratch/$USER/
+  git clone https://github.com/GridMod/MSPCM-Workshop.git
+  cd MSPCM-Workshop/Workshop-Explorations/OneWeek
+  ```
+
+2. Within `OneWeek` we will create a new environment file [`newenv`](RunFiles/newenv) using the following commands
+  ```bash
+  echo 'export PLEXOS_VERSION="8.0"
+  export XPRESSMP_VERSION="8.5.6"
+  module purge
+  module load centos mono/4.6.2.7 xpressmp/$XPRESSMP_VERSION plexos/$PLEXOS_VERSION
+  module load conda
+
+  export PLEXOS_TEMP=/scratch/$USER/tmp/$SLURM_JOBID
+  export TEMP=$PLEXOS_TEMP
+  mkdir -p $PLEXOS_TEMP
+  ' > newenv
+  ```
+  This will ensure that the appropriate modules and conda versions are loaded and
+  an output directory.
+
+3. Request an interactive session on Eagle
+
+  ```bash
+  srun --account=MYACCOUNT --time=01:00:00 --partition=debug --ntasks=1 --nodes=1 --pty bash
+  ```
+
+4. Setup the new environment. Make sure that you have `numpy` and `pandas` in your
+`plex1` environment.
+
+  ```bash
+  # Ensure that we are in the correct working directory
+  cd /scratch/$USER/MSPCM-Workshop/Workshop-Explorations/OneWeek
+  source newenv
+  conda activate plex1
+  ```
+
+5. Run the [`get_week.py`](RunFiles/get_week.py) script. This script essentially extracts a weeks worth of
+data for the simulation
+
+  ```bash
+  python get_week.py
+  ```
+
+6. Finally we run our day ahead example using the following command
+
+  ```bash
+  mono $PLEXOS/PLEXOS64.exe -n "one_week_model.xml" -m DAY_AHEAD
+  ```
+
+## Example 3: Simple Batch Job Submission
+
+Up till now we have been running PLEXOS examples in an interactive settings. Now
+we will try to submit a batch job into the Eagle queue. This job uses the same
+`one_week_model.xml` file from Example 2, but for the sake of completness, we
+will download a tarball of the same from a different repo in our batch script.
+
+1. Get batch script by running the command
+
+  ```bash
+  wget https://github.nrel.gov/raw/tkaiser2/plexos/master/scripts/simple
```

Comment:
Instead of pulling the script, I think it would be better to just say how to run it from this repo. What do you think?

Reply:
So the file already exists in the repo, I can change the wording to point to the one in the repo. My main concern would be where the latest working version of the file  will be maintained. I agree that it should be maintained in this repo but that had not been the case before.